libs/execution_context/ebpf_core.c

```diff
@@ -1324,6 +1327,46 @@ _ebpf_core_get_time_ns()
     return ebpf_query_time_since_boot(false) * 100;
 }
 
+// Pick a limit on string size based on the size of the eBPF stack.
+#define MAX_PRINTK_STRING_SIZE 512
+
+long
+_ebpf_core_trace_printk(_In_reads_(fmt_size) const char* fmt, size_t fmt_size)
+{
+    if (fmt_size > MAX_PRINTK_STRING_SIZE - 1) {
+        // Disallow large fmt_size values.
+        return -1;
+    }
+
+    // Make a copy of the original string.
+    char* output = (char*)ebpf_allocate(fmt_size + 1);
+    if (output == NULL) {
+        return -1;
+    }
+    memcpy(output, fmt, fmt_size);
+
+    // Make sure the output is null-terminated.
+    // Remove the newline if not already present.
```

Comment:
This comment doesn't parse for me. Was this supposed to be:
Remove the newline if present.

Reply:
Oops, yes :)